src/libs/decoders/opus.cpp

```diff
@@ -266,39 +281,38 @@ static uint32_t opus_read(Sound_Sample * sample, void * buffer, uint32_t request
     assertm(sample && buffer, "OPUS: Inputs are not initialized [bug]");
     if (requested_frames == 0)
         return 0u;
-    Sound_SampleInternal* internal =
-        reinterpret_cast<Sound_SampleInternal*>(sample->opaque);
-    OggOpusFile* of =
-        reinterpret_cast<OggOpusFile*>(internal->decoder_private);
-    opus_int16 *sample_buffer =
-        reinterpret_cast<opus_int16*>(buffer);
 
+    Sound_SampleInternal* internal = static_cast<Sound_SampleInternal*>(sample->opaque);
+    OggOpusFile* of = static_cast<OggOpusFile*>(internal->decoder_private);
     const uint32_t channels = sample->actual.channels;
-    const uint32_t requested_samples = requested_frames * channels;
-    uint32_t decoded_samples = 0;
-    int result = 0;
-    while(decoded_samples < requested_samples) {
-        result = op_read(
-            of,                                  // pointer to the opusfile object
-            sample_buffer + decoded_samples,     // buffer offset to save decoded samples
-            requested_samples - decoded_samples, // remaining samples to decode
-            nullptr);                               // link index, which we don't use
-        // Check the result code
+
+    // Initial state-tracking variables
+    uint32_t total_decoded_samples = 0;
+    opus_int16 *buf_pos = static_cast<opus_int16*>(buffer);
+    int32_t remaining_samples = static_cast<int32_t>(requested_frames * channels);
+
+    // Start the decode loop, incrementing as we go
+    while(remaining_samples > 0) {
+        const int result = op_read(of, buf_pos, remaining_samples, nullptr);
```

Comment:
Hey, that's the suggestion I made earlier in this review :) Nice!

Reply:
:+1: